### PR TITLE
abinit: Add version 9.6.1 and support fujitsu compiler and fujitsu-fftw.

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ class Abinit(AutotoolsPackage):
     # using `conflicts`, `depends_on` could be used instead.
     mkl_message = 'Need to set dependent variant to threads=openmp'
     conflicts('+openmp',
-              when='^fftw~openmp',
+              when='^fftw-api~openmp',
```
Comment (haampie) [inline var/spack/repos/builtin/packages/abinit/package.py]: There is no defined interface for `fftw-api`, so you can't assume you can set `~openmp` on it. You'll need to do that per implementation now.

Comment (tkameyama) [reply var/spack/repos/builtin/packages/abinit/package.py]: fixed.

Comment (tkameyama) [reply var/spack/repos/builtin/packages/abinit/package.py]: @haampie
Do I need to make other changes?

**Comment by tgamblin** [comment]
@haampie can you have another look at this?

**Review by tgamblin — APPROVED** [review]

**Pull request merged**